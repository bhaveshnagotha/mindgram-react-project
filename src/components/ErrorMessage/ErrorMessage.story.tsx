import React from 'react'

import { text, withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'

import ErrorMessage from '.'

storiesOf('Error', module)
  .addDecorator(withKnobs)
  .add('ErrorMessage', () => (
    <ErrorMessage
      message="Something went wrong!"
      color={text('Color', 'red')}
    />
  ))
