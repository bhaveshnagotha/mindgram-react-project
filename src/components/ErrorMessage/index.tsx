import React, { Component } from 'react'

interface IProps {
  message?: string
  color?: string
}
class ErrorMessage extends Component<IProps> {
  private static defaultProps = {
    color: 'red',
    message: 'Under construction',
  }

  public render() {
    const { message, color } = this.props
    return <div style={{ color }}>{message}</div>
  }
}

export default ErrorMessage
