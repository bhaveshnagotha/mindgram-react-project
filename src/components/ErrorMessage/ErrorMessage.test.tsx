import React from 'react'

import { shallow } from 'enzyme'

import ErrorMessage from '.'

const defaultProps = {}

function shallowRender(newProps = {}) {
  const props = {
    ...defaultProps,
    ...newProps,
  }
  return shallow(<ErrorMessage {...props} />)
}

describe('<ErrorMessage />', () => {
  it('matches default snapshot', () => {
    const wrapper = shallowRender()
    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot', () => {
    const wrapper = shallowRender({
      message: 'Whoops!',
    })
    expect(wrapper).toMatchSnapshot()
  })
})
