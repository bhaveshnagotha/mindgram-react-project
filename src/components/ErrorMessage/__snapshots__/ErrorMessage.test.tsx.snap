// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> matches default snapshot 1`] = `ShallowWrapper {}`;

exports[`<ErrorMessage /> matches snapshot 1`] = `ShallowWrapper {}`;
